django/db/models/options.py

```diff
@@ -711,6 +711,17 @@ def get_parent_list(self):
                 result.add(ancestor)
         return list(result)
 
+    def get_non_abstract_parent_list(self):
+        """
+        Return all the non abstract ancestors of this model as a list.
+        """
+        parent_list = self.get_parent_list()
+        # Filter the list to exclude abstract models
+        concrete_parents = [
+            parent for parent in parent_list if not parent._meta.abstract
+        ]
+        return concrete_parents
```

Comment:
We don't use this list, so there is no need to create it, maybe
```suggestion
    def has_non_abstract_parent(self):
        return any(not parent._meta.abstract for parent in self.get_parent_list())
```
On the other hand to we need a separate helper for one-liner that is only used in one place?

Reply:
> On the other hand do we need a separate helper for one-liner that is only used in one place?

+1 too many abstractions can hinder maintenance

Reply:
I like felix's `any()` variant too: Not only does it look more performant at a glance but it also reads better than generating a list then checking its length.